vm/sdk/src/spawn.rs

```diff
@@ -0,0 +1,55 @@
+// Template address is read from context
+pub fn spawn(blob: Vec<u8>) {
+  let blob_u32 = bytes_to_u32_vec(&blob);
+  athena_vm::syscalls::spawn(blob_u32.as_ptr(), blob.len());
+}
+
+fn bytes_to_u32_vec<T: AsRef<[u8]>>(bytes: T) -> Vec<u32> {
+  let mut chunks = bytes.as_ref().chunks_exact(4);
+  let mut result = chunks
+    .by_ref()
+    .map(|chunk| {
+      let mut bytes = [0u8; 4];
+      bytes.copy_from_slice(chunk);
+      u32::from_le_bytes(bytes)
+    })
+    .collect::<Vec<u32>>();
+
+  let remainder = chunks.remainder();
+  if !remainder.is_empty() {
+    let mut bytes = [0u8; 4];
+    bytes[..remainder.len()].copy_from_slice(remainder);
+    result.push(u32::from_le_bytes(bytes));
+  }
+
+  result
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+
+  #[test]
+  fn convert_empty_slice() {
+    let result = bytes_to_u32_vec([]);
+    assert_eq!(result, Vec::new());
+  }
+
+  #[test]
+  fn convert_exact_multiple_of_4() {
+    let result = bytes_to_u32_vec([1, 2, 3, 4, 5, 6, 7, 8]);
+    assert_eq!(result, vec![0x04030201, 0x08070605]);
+  }
+
+  #[test]
+  fn covnert_not_a_multiple_of_4() {
+    let result = bytes_to_u32_vec([1, 2, 3, 4, 5, 6, 7]);
+    assert_eq!(result, vec![0x04030201, 0x00070605]);
+  }
+
+  #[test]
+  fn convert_single_byte() {
+    let result = bytes_to_u32_vec([1]);
+    assert_eq!(result, vec![0x01]);
```

Comment:
This is a bit counterintuitive, worth adding a comment explaining what's going on here

Reply:
Why is it counterintuitive? I updated the test to show that the u32 is a zero-padded little-endian value and added a doc comment on `bytes_to_u32_vec`.